src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp

```diff
@@ -2356,6 +2356,28 @@ struct AttentionExecutor : public PagedAttentionExecutor {
                 _slot_mapping.ptr<int32_t>()[idx++] =
                     block_number * _helper._block_size + block_offset % _helper._block_size;
             }
+            // To simplify tails of the kernels for Q*K and W*V:
+            // for first token kernels:
+            //    Q*K aka [m, k0] * [n0, k0]', there is already tails logic for m, k0, but no(always assume n0 is
+            //    block_size) for n0 which means the tail of k_cache need to be set to zero.
+            //    W*V aka [m, k1] * [n1, k1]', there is no tails handing for n1, so tails of v_cache need to be set to
+            //    zero.
+            // for second token, the kernels have tails handling logic
+            if (q_len != 1 && kv_len % _helper._block_size != 0) {
```

Comment:
So in serving scenarios (where prompt processing is interleaved with second token generation) and in beam-search or speculative decoding cases (where even seconf token processed with q != 1) we will have memsets on each iteration? If so that might be usefull to directly state it in the comments.

Reply:
In serving scenarios or beam-search, there still needs to be processed one prompt by one due to each kv-cache is not shared. So in the mixed first and second scenarios, the first tokens will be picked up and memset, the second tokens will be kept unchanged.